yt/frontends/ramses/data_structures.py

```diff
@@ -666,6 +686,29 @@ def __init__(self, filename, dataset_type='ramses',
 
         self.storage_filename = storage_filename
 
+        # Add particles filters
+        if ('io', 'particle_family') in self.field_list:
```

Comment:
Putting this line inside of `__init__` leads to some unintended consequences that I think is triggering the failures. Normally in yt field detection (and thus the creation of the `field_list`) happens when `ds.index` is created. This normally does not happen when we initially load a dataset (and thus when `__init__` is called). However, by invoking `self.field_list` here you are triggering field detection and the creation of the index. But, below, you made it so that the particle filters only get registered during field detection (by adding an implementation of `_create_field_info`). This means that your particle filters never actually get set up, because by the time you register them with `add_particle_filter` a few lines below here, field detection has already hapenned, and we've already failed to add the particle filters.

The fix is to move this check out of `__init__` and into `_create_field_info`. Specifically, I'd suggest applying the following diff:

```
diff --git a/yt/frontends/ramses/data_structures.py b/yt/frontends/ramses/data_s
tructures.py
index c19406b35..ebb6781ba 100644
--- a/yt/frontends/ramses/data_structures.py
+++ b/yt/frontends/ramses/data_structures.py
@@ -686,7 +686,11 @@ class RAMSESDataset(Dataset):

         self.storage_filename = storage_filename

-        # Add particles filters
+
+    def create_field_info(self, *args, **kwa):
+        """Extend create_field_info to add the particles types."""
+        super(RAMSESDataset, self).create_field_info(*args, **kwa)
+        # Register particle filters
         if ('io', 'particle_family') in self.field_list:
             for fname, value in particle_families.items():
                 def loc(val):
@@ -697,15 +701,6 @@ class RAMSESDataset(Dataset):
                     return closure
                 add_particle_filter(fname, loc(value),
                                     filtered_type='io', requires=['particle_family'])
-
-
-    def create_field_info(self, *args, **kwa):
-        """Extend create_field_info to add the particles types."""
-        Dataset.create_field_info(self, *args, **kwa)
-        self._add_ptypes()
-
-    def _add_ptypes(self):
-        if ('io', 'particle_family') in self.field_list:
             for k in particle_families.keys():
                 mylog.info('Adding particle_type: %s' % k)
                 self.add_particle_filter('%s' % k)
```

Does that make sense?